src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -247,43 +273,46 @@ function compute_gwd_fields(domain) result(iErr)
 
          !
          ! Cut out a rectangular piece of the global 30-arc-second topography
-         ! data that is centered at the lat/lon of the current cell being
+         ! data that is centered at the lat/lon (in radians) of the current cell being
          ! processed and that is just large enough to cover the cell. The
          ! rectangular array of topography data is stored in the module
-         ! variable 'box', and the dimensions of this array are given by the
-         ! module variables 'nx' and 'ny'. The get_box() routine also
+         ! variable 'box', and the dimensions of this array are obtained from 
+         ! the routine get_box_size_from_lat_lon and store in the
+         ! local variables 'nx' and 'ny'. The get_box() routine also
          ! computes the mean elevation in the array and stores that value in
          ! the module variable 'box_mean'.
          !
-         call get_box(latCell(iCell)*rad2deg, lonCell(iCell)*rad2deg, dc)
+         call get_box_size_from_lat_lon(latCell(iCell)*rad2deg, lonCell(iCell)*rad2deg, dc, nx, ny)
+         call get_box(latCell(iCell)*rad2deg,lonCell(iCell)*rad2deg, nx, ny, geog_data_path, geog_sub_path, tilesHead)
 
          !
          ! With a box of 30-arc-second data for the current grid cell, call
          ! subroutines to compute each sub-grid orography statistic
          !
-         var2d(iCell) = get_var()
-         con(iCell) = get_con()
-         oa1(iCell) = get_oa1()
-         oa2(iCell) = get_oa2()
-         oa3(iCell) = get_oa3()
-         oa4(iCell) = get_oa4()
+         var2d(iCell) = get_var(nx, ny)
+         con(iCell) = get_con(nx, ny)
+         oa1(iCell) = get_oa1(nx, ny)
+         oa2(iCell) = get_oa2(nx, ny)
+         oa3(iCell) = get_oa3(nx, ny)
+         oa4(iCell) = get_oa4(nx, ny)
 
          ! Critical height, to be used in OL computation
          ! See Appendix of Kim, Y-J, 1996: Representation of Sub-Grid Scale Orographic Effects
          ! in a General Circulation Model. J. Climate, 9, 2698-2717.
          hc = 1116.2_RKIND - 0.878_RKIND * var2d(iCell)
 
-         ol1(iCell) = get_ol1()
-         ol2(iCell) = get_ol2()
-         ol3(iCell) = get_ol3()
-         ol4(iCell) = get_ol4()
+         ol1(iCell) = get_ol1(nx, ny)
+         ol2(iCell) = get_ol2(nx, ny)
+         ol3(iCell) = get_ol3(nx, ny)
+         ol4(iCell) = get_ol4(nx, ny)
 
-         hlanduse(iCell) = get_dom_landmask()  ! get dominant land mask in cell
+         hlanduse(iCell) = get_dom_landmask(nx, ny)  ! get dominant land mask in cell
 
-!         elvmax(iCell) = get_elvmax()
-!         htheta(iCell) = get_htheta()
-!         hgamma(iCell) = get_hgamma()
-!         hsigma(iCell) = get_hsigma()
+         ! elvmax_1 = get_elvmax(nx,ny)
+         ! htheta_1 = get_htheta(nx,ny)
+         ! hgamma_1 = get_hgamma(nx,ny)
+         ! hsigma_1 = get_hsigma(nx,ny)
```

Comment:
I'd recommend keeping this code as it was before.

Reply:
done, but indenting the lines to match the immediate surroundings, as we discussed.